[ylylma] iP

ylyma opened the pull request:
# Kiera

> "Your mind is for having ideas, not holding them." -David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Kiera frees your mind of having to remember things you need to do. With Kiera, you can effortlessly keep track of your tasks and stay organized like never before.

🚀 **Features:**
- [X] 📝 Managing tasks
- [X] 🔎 Search and filter by date or keyword
- [ ] ⏰ Reminders (coming soon)

💻 **Getting Started**
1. Download the .jar file from [here](https://github.com/ylyma/ip/releases/tag/Level-9)
2. Double-click it
3. Add your tasks!

📣 **Usage**
* Add tasks
   ```
   todo task1
   deadline task2 /by 2023-09-04
   event task3 /from 2023-09-04 1300 /to 1500
   ```
* List tasks
   ```
   list
   ```
* Delete tasks by number
  ```
  delete 2
  ```
* Find tasks by keyword
  ```
  find task
  ```
* Filter tasks by date
  ```
  deadline-date 2023-09-04
  deadline-today
  event-date 2023-09-04
  event-today
  ```
-----
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ylyma committed:
rename, greet, exit

ylyma committed:
rename, greet, exit

ylyma committed:
echo

ylyma committed:
add, list + change tone

ylyma committed:
mark as done

ylyma committed:
added classes

ylyma committed:
todos, events, deadlines

ylyma committed:
exceptions

ylyma committed:
delete

ylyma committed:
ui testing

ylyma committed:
Add save file and functions

ylyma committed:
Add save file and functions

ylyma committed:
Add date and time commands and objects

ylyma committed:
Add more classes

ylyma committed:
Add enum TaskTypes

ylyma committed:
Organise into packages

ylyma committed:
Merge branch 'add-gradle-support'

ylyma committed:
Add Gradle files

ylyma committed:
Add 2 JUnit tests

ylyma committed:
Create JAR file

ylyma committed:
Add Javadocs

ylyma committed:
Follow coding standards

ylyma committed:
Merge branch 'branch-A-Javadoc'

ylyma committed:
Add Javadoc (#1)

* Add Gradle support

* Bump gradle and lib version

* Add Gradle files

* Add 2 JUnit tests

* Create JAR file

* Add Javadocs

---------

Co-authored-by: damithc <damithch@damithch-mbp.comp.nus.edu.sg>
Co-authored-by: lzq <li.zhaoqi@dhs.sg>

ylyma committed:
Branch a coding standard (#2)

* Add Gradle support

* Bump gradle and lib version

* Add Gradle files

* Add 2 JUnit tests

* Create JAR file

* Add Javadocs

---------

Co-authored-by: damithc <damithch@damithch-mbp.comp.nus.edu.sg>
Co-authored-by: lzq <li.zhaoqi@dhs.sg>

ylyma committed:
Add find by keyword function

ylyma committed:
Resolve merge conflicts.

ylyma committed:
Merge branch 'branch-Level-9'

ylyma committed:
Resolve merge conflicts

ylyma committed:
Merge branch 'branch-Level-9'

ylyma committed:
Update README.md

ylyma committed:
Add checkStyle

ylyma committed:
Add CheckStyle

ylyma committed:
Merge branch 'master' of https://github.com/ylyma/ip

RSXIX commented on src/main/java/kiera/command/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+import kiera.task.Task;
+
+/**
+ * Represents a command for deleting a tasl from tha taskList.
```
Think you mis spelled task here

RSXIX commented on src/main/java/kiera/command/ListCommand.java:
```diff
@@ -0,0 +1,37 @@
+        if (tasks.isEmpty()) {
+            System.out.println("nothing for you to do yet!");
+        }
+        String result = tasks.getTasks().stream()
```
Neat use of streams

RSXIX reviewed (commented):
The code felt very succient and well thought out. Good use of readable variables and little to none violation of coding standards.

ylyma committed:
Add GUI for application

ylyma committed:
Correct spelling

Nauxe commented on src/main/java/kiera/Kiera.java:
```diff
@@ -0,0 +1,64 @@
+    }
+}
+
+//    public void run() {
+//        ui.showHello();
+//        boolean isExit = false;
+//        while (!isExit) {
+//            try {
+//                String fullCommand = ui.readCommand();
+//                ui.showLine();
+//                Command c = Parser.parse(fullCommand);
+//                c.execute(tasks, ui, storage);
+//                isExit = c.isExit();
+//            } catch (KieraException e) {
+//                ui.showError(e.toString());
+//            } finally {
+//                ui.showLine();
+//            }
+//        }
+//    }
+
```
I think you can remove this section of code or stash it locally

Nauxe commented on src/main/java/kiera/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        }
+
+        if (input.startsWith("deadline-today")) {
+            String desc = LocalDate.now().toString();
```
Good idea to let users create deadlines that are due on the current day!

Nauxe reviewed (commented):
LGTM! Overall very clean code! I agree with the other review about using stream and filter, which makes the code look extremely neat. Your bot also has a really cute personality! ^-^

ylyma committed:
Add assertions to DeleteCommand.

DeleteCommand can only occur if the task list is not empty.

Adding assertion to ensure that task list is not empty ensures that the
DeleteCommand is only called when there is something to delete.

Let's add the assertion that the task list is not null to make sure that
the assumption is fulfilled before command is executed.

ylyma committed:
Add assertions for MarkCommand

ylyma committed:
Add assertions (#3)

* Add assertions to DeleteCommand.

DeleteCommand can only occur if the task list is not empty.

Adding assertion to ensure that task list is not empty ensures that the
DeleteCommand is only called when there is something to delete.

Let's add the assertion that the task list is not null to make sure that
the assumption is fulfilled before command is executed.

* Add assertions for MarkCommand

ylyma committed:
Remove dead code and add Javadocs

ylyma committed:
Remove dead code and add Javadocs (#4)

ylyma committed:
Add CI

ylyma committed:
Add SortCommand class

There is no way to sort tasks by date or time.

Added a SortCommand class with methods that allow for the sorting of
deadlines and events in ascending order of date or time.

Users will most likely want to visualise which deadlines or events are
coming soon.

Let's allow the sorting of tasks with a command.

ylyma committed:
Add CI

ylyma committed:
Edit CI

ylyma committed:
Change import order

ylyma committed:
Update tests

ylyma committed:
Merge branch 'branch-A-CI' of https://github.com/ylyma/ip into branch-C-sort

ylyma committed:
Add CI (#8)

* Remove dead code and add Javadocs

* Add CI

* Add CI

* Edit CI

* Change import order

* Update tests

ylyma committed:
Resolve merge conflicts

ylyma committed:
Change import order

ylyma committed:
Correct style errors

ylyma committed:
Add SortCommand (#9)

* Remove dead code and add Javadocs

* Add CI

* Add SortCommand class

There is no way to sort tasks by date or time.

Added a SortCommand class with methods that allow for the sorting of
deadlines and events in ascending order of date or time.

Users will most likely want to visualise which deadlines or events are
coming soon.

Let's allow the sorting of tasks with a command.

* Add CI

* Edit CI

* Change import order

* Update tests

* Change import order

* Correct style errors

ylyma committed:
Merge branch 'master' of https://github.com/ylyma/ip

ylyma committed:
Update gitignore

ylyma committed:
Edit build.gradle

ylyma committed:
Remove binary files

ylyma committed:
Add user guide (#11)

ylyma committed:
Edit user guide
